datafusion/expr-common/src/signature.rs

```diff
@@ -365,7 +365,12 @@ impl TypeSignature {
         }
     }
 
-    /// get all possible types for the given `TypeSignature`
+    /// This function is used specifically internally for `information_schema`
+    /// We suggest not to rely on this function
+    ///
+    /// Get all possible types for `information_schema` from the given `TypeSignature`
+    //
+    // TODO: Make this function private
```

Comment:
we could potentially mark it as deprecated and make the new (non pub) version with a different name

Maybe something like this (can do in a follow on PR)

```rust
    pub fn get_example_types(&self) -> Vec<Vec<DataType>> {
```

Reply:
I think we would need NativeType for `get_example_types`